<launch>

    <arg name="prefix" default=""/>
    
    <param name="robot_description" 
        textfile="$(find coman_urdf)/urdf/coman.urdf"/>
    
    <param name="robot_description_semantic"
        textfile="$(find vsam)/srdf/coman.srdf"/>
    
    <param name="cartesian/problem_description" 
        textfile="$(find vsam)/configs/coman_stack_vs_rg.yaml"/>
    
    <include file="$(find cartesian_interface)/launch/cartesio.launch">
        <arg name="rate" value="1000.0"/>
        <arg name="prefix" value="$(arg prefix)"/>
    </include>

    <node pkg="opensot_visual_servoing" type="image_processing_rg_node" name="image_processing_rg_node">
	
	<remap from="camera_info" to="camera/rgb/camera_info"/>
    <remap from="image_raw" to="camera/rgb/image_raw"/>
    <remap from="image_processing/visual_features" to="cartesian/visual_servoing_camera_link/features"/>

	<remap from="desired_visual_features" to="reference_governor/desired_features"/>	
	<remap from="image_processing/reference_features" to="cartesian/visual_servoing_camera_link/desired_features"/>	
	
	</node> 

	 <node pkg="ReGoVis" type="reference_governor.py" name="reference_governor">

	 <remap from="reference_governor/reference_features" to="cartesian/visual_servoing_camera_link/desired_features"/>
	 </node>

	<rosparam param="regovis_MPC_sampling_time">0.1</rosparam>
	<rosparam param="regovis_preview_window_size">10</rosparam> 
	<rosparam param="image_processing/depth">4</rosparam>
    
</launch>
